ext/ddtrace_profiling_native_extension/profiling.c

```diff
@@ -1,6 +1,7 @@
 #include <ruby.h>
 #include <ruby/thread.h>
 #include <errno.h>
+#include <malloc.h>
```

Comment:
This include will need guarding as well otherwise we break compilation in macos:

```
../../../../ext/ddtrace_profiling_native_extension/profiling.c:4:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
```

Reply:
Ooops! Thanks for catching it, fixed in b2a487943a5a69911b235f07d66299af4f2f7dab.

(Out of curiosity, this wasn't caught by macOS CI because we still don't build the profiler on macOS, since it's a configuration only used for development).